service-go/internal/api/api.go

```diff
@@ -1,23 +1,67 @@
 package api
 
 import (
+	"cloud.google.com/go/pubsub"
 	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"time"
 
 	"interview-service/internal/api/interview"
 	"interview-service/internal/domain/greeter"
 )
 
 type server struct {
 	interview.UnimplementedInterviewServiceServer
+	resultProjectID string
+	resultTopicID   string
 }
 
-func New() *server {
-	return &server{}
+func New(projectID, topicID string) *server {
+	return &server{
+		resultProjectID: projectID,
+		resultTopicID:   topicID,
+	}
 }
 
 func (s *server) HelloWorld(ctx context.Context, r *interview.HelloWorldRequest) (*interview.HelloWorldResponse, error) {
 	greeting := greeter.Greet(r.GetName())
 
+	// simulate some async work with pubsub here
+	go func() {
```

Comment:
Goroutine to simulate some async work occurring on the server. A goroutine is used to avoid delaying the http response. This is a very common pattern in the cloud native services I work on. This code would also need additional analysis and testing for error and exit handling before it could be used in a production scenario.